### PR TITLE
Make index serial scan concurrency configurable and fix golint errors.

### PR DESCRIPTION
**Pull request by shenli**
When doing index double read operation with keepOrder as true, we use only one worker. If we need to read a lot of data, it will be quite slow. So I make it configurable. For OLAP TiDB cluster, we could set the concurrency to 4 or 5.

I will add test in the next commit.

### CODE REVIEW

File: sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -70,6 +70,10 @@ const (
 	// Set this value higher may reduce the latency but consumes more system resource.
 	TiDBIndexLookupConcurrency = "tidb_index_lookup_concurrency"
 
+	// tidb_index_serial_scan_concurrency is user for controling the concurrency of index scan operation
```
Comment (coocood) [inline sessionctx/variable/tidb_vars.go]: is used

**Comment by shenli** [comment]
@coocood @tiancaiamao @zimulala PTAL

**Comment by ngaut** [comment]
Hi, do we have any bench results ?

**Comment by shenli** [comment]
@ngaut In one of SMJ case, I increase the concurrency from 1 to 3 and query time drop from 2min to 1min 30sec. But this will cost more memory.

**Comment by ngaut** [comment]
Cool

**Pull request merged**